src/app/pages/quizzes-catalog/quizzes-catalog.component.ts

```diff
@@ -1,25 +1,43 @@
-import { Component } from '@angular/core';
+import { Component, OnDestroy, OnInit } from '@angular/core';
 import { UIKitModule } from '../../ui-kit/ui-kit.module';
-import { quizzesCatalog } from '../../test-data/quizzes-catalog';
 import { CommonModule } from '@angular/common';
 import {
   CardColors,
   CardItemStyle,
   cardItemStyles,
 } from '../../ui-kit/constants/card-item';
+import { CategoryModel } from '../../services/model/category.model';
+import { Subscription } from 'rxjs';
+import { CategoryService } from '../../services/catalog.service';
 
 @Component({
   standalone: true,
   selector: 'app-quizzes-catalog',
   templateUrl: './quizzes-catalog.component.html',
   imports: [ UIKitModule, CommonModule ],
 })
-export class QuizzesCatalogComponent {
-  quizzesCatalog = quizzesCatalog;
+export class QuizzesCatalogComponent implements OnInit, OnDestroy {
+  quizzesCatalog: CategoryModel[] = [];
+  subscription: Subscription = new Subscription();
```

Comment:
new Subscription() we use if we want to group multiple subscription like 
```
this.subscription.add(something1.subscribe()).
this.subscription.add(something2.subscribe()).
```

When you manage only one Subscription, you can omit initializing it, so here you can use just `subscription: Subscription`, without `= new Subscription();`